native/src/routes/JpalTracking.tsx

```diff
@@ -30,43 +32,94 @@ const DescriptionContainer = styled.TouchableOpacity`
   justify-content: space-between;
   padding: 15px 0 5px;
 `
-const Input = styled(TextInput)`
+const Input = styled(TextInput)<{ editable: boolean; error: boolean }>`
   padding: 15px;
   border-width: 1px;
   border-color: ${props => (props.editable ? props.theme.colors.themeColor : props.theme.colors.textDisabledColor)};
+  ${props => (props.error ? 'border-color: red;' : '')};
   color: ${props => props.theme.colors.textColor};
   text-align-vertical: top;
   height: 50px;
 `
 
 export type PropsType = {
-  theme: ThemeType
   route: RoutePropType<JpalTrackingRouteType>
   navigation: NavigationPropType<JpalTrackingRouteType>
 }
 
-const JpalTracking = ({ route, theme }: PropsType) => {
+const TRACKING_CODE_LENGTH = 7
+
+const JpalTracking = ({ navigation, route }: PropsType): ReactElement => {
   const [trackingCode, setTrackingCode] = useState<string | null>(null)
   const [trackingEnabled, setTrackingEnabled] = useState<boolean | null>(null)
-  const [settingsLoaded, setSettingsLoaded] = useState(false)
-  const [error, setError] = useState<boolean>(false)
+  const [settingsLoaded, setSettingsLoaded] = useState<boolean>(false)
+  const [invalid, setInvalid] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
   const { t } = useTranslation(['settings', 'error'])
+  const theme = useTheme()
   const routeTrackingCode = route.params.trackingCode
 
-  const updateTrackingCode = (value: string) => {
+  const updateTrackingEnabled = useCallback((trackingEnabled: boolean) => {
+    setTrackingEnabled(trackingEnabled)
+    appSettings
+      .setJpalTrackingEnabled(trackingEnabled)
+      .then(() => setError(null))
+      .catch(() => setError('generalError'))
+  }, [])
+
+  const toggleTrackingEnabled = () => {
+    updateTrackingEnabled(!trackingEnabled)
+  }
+
+  const updateTrackingCode = useCallback((value: string) => {
     setTrackingCode(value)
     appSettings
       .setJpalTrackingCode(value)
-      .then(() => setError(false))
-      .catch(() => setError(true))
-  }
+      .then(() => setError(null))
+      .catch(() => setError('generalError'))
+  }, [])
 
+  useEffect(
+    () =>
+      navigation.addListener('beforeRemove', e => {
+        if (!routeTrackingCode) {
+          // User probably just opened the settings out of interest, don't annoy em by opening an alert
+          return
+        }
+        if (trackingCode?.length === TRACKING_CODE_LENGTH && trackingEnabled) {
```

Comment:
So it you don't have a trackingCode in the url, it is fine to enter a TrackingCode with wrong lenght? Not sure if this is really wanted behaviour. Also would it make sense to give the user some feedback what the correct lenght would be?